sklearn/mixture/gmm.py

```diff
@@ -352,6 +352,27 @@ def predict(self, X):
         logprob, responsibilities = self.score_samples(X)
         return responsibilities.argmax(axis=1)
 
+    def fit_predict(self, X, y=None):
+        """
+        Fit and then predict labels for data.
+        Warning: due to the final maximization step in the EM algorithm,
+        with low iterations the prediction may not be 100% accurate
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+
+        Returns
+        -------
+        C : array, shape = (n_samples,)
+        """
+        self.fit(X, y)
+
+        if self.responsibilities_ is None:
+            raise RuntimeError("Fitting failed, cannot predict")
```

Comment:
It's better to use the `_check_fitted_model`: do a `git grep _check_fitted_model` to see examples in the code base.